### PR TITLE
[angular] upgrade to angular to v14

### PR DESCRIPTION
**Pull request by mshima**
Fix https://github.com/jhipster/generator-jhipster/issues/18817
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
Blocked on https://github.com/just-jeb/angular-builders/issues/1205.
Will happen for Angular v14 only.

**Comment by henryruhs** [comment]
"Will happen for Angular v14 only."

It would be nice to have a next release - there are libraries that properly depend on this and should have the chance to test against Angular 14 release candidates to have support from day one of the final release.

File: generators/client/templates/angular/src/main/webapp/app/login/login.component.ts.ejs [hunk 1]
```diff
@@ -33,18 +33,13 @@ export class LoginComponent implements OnInit, AfterViewInit {
 
   authenticationError = false;
 
-  loginForm = this.fb.group({
-    username: [null, [Validators.required]],
-    password: [null, [Validators.required]],
-    rememberMe: [false],
+  loginForm = new FormGroup({
+    username: new FormControl('', { nonNullable: true, validators: [Validators.required] }),
```
Comment (Tcharl) [inline generators/client/templates/angular/src/main/webapp/app/login/login.component.ts.ejs:37]: isn't it risky?

Comment (Tcharl) [reply generators/client/templates/angular/src/main/webapp/app/login/login.component.ts.ejs:37]: Replacing null value par en empty string doesn't mean the same thing: 
 Here, for example the required validator is useless...

Comment (mshima) [reply generators/client/templates/angular/src/main/webapp/app/login/login.component.ts.ejs:37]: In this case `''` is used for the type, `''` and `null` are treated equally.
Example if we set `null`, the user types something and deletes, we have `''` or `null`? So they are the same.
`''` is used just for the type (string).

**Comment by DanielFran** [comment]
@mshima Is there any issue on [ng-bootstrap](https://github.com/ng-bootstrap/ng-bootstrap/issues) that we could track?

**Comment by mshima** [comment]
@DanielFran i suppose it’s a known issue due to the announcement https://github.com/ng-bootstrap/ng-bootstrap/issues/4340#issuecomment-1155208704

**Comment by mshima** [comment]
There will be too many changes and became hard to keep everything consistent and test passing.
I will split into several small PRs.

**Comment by mraible** [comment]
Can you please rebase to fix conflicts, @mshima?

**Comment by DanielFran** [comment]
@mshima @mraible Seems that ng-bootstrap will be released today, so I can release a new version soon that will include it

**Comment by mshima** [comment]
Please squash when merging.

**Review by qmonmert — APPROVED** [review]
LGTM 👏

**Comment by mshima** [comment]
@DanielFran bounty claimed https://opencollective.com/generator-jhipster/expenses/102422.

**Comment by DanielFran** [comment]
@mshima approved

**Pull request merged**